### ISSUE TITLE
MAX_STRING_SIZE doesn't work with image

### ISSUE THREAD

**Issue by CoreyKaylor** [oracle/docker-images#399]
I've tried "basic" and "extreme" for the DB_BUNDLE setup and followed these steps to configure the system settings.

```
# Upgrade oracle to allow long varchar types
SQL> show parameter string_size
NAME TYPE VALUE
------------------------------------ ----------- ------------------------------
max_string_size string STANDARD

SQL> alter system set max_string_size=EXTENDED scope=spfile;

SQL> shutdown immediate;

SQL> startup upgrade;

SQL> @pathtodir/rdbms/admin/utl32k.sql

SQL> shutdown immediate;

SQL> startup;

SQL> show parameter string_size

NAME TYPE VALUE
------------------------------------ ----------- ------------------------------
max_string_size string EXTENDED
```

Even after it reports that EXTENDED is configured I cannot create any tables that exceed the default max length and still get `ORA-00910: specified length too long for its datatype`

**Comment by CoreyKaylor** [comment]
I also followed the steps outlined in https://www.markusdba.net/?p=197 with no luck either. Any help would be appreciated.

**Comment by gvenzl** [comment]
Hi @CoreyKaylor, have you also run the script on the PDB level?

**Comment by CoreyKaylor** [comment]
@gvenzl yes as outlined in the blogpost I linked to. Changing the applicable pdb names to the ones inside the container.

**Comment by gvenzl** [comment]
I just tested it following the steps in the [documentation](http://docs.oracle.com/database/122/REFRN/MAX_STRING_SIZE.htm#REFRN10321) and it worked fine for me.

```
[oracle@285659826a0c ~]$ sqlplus / as sysdba

SQL*Plus: Release 12.2.0.1.0 Production on Tue May 30 19:40:36 2017

Copyright (c) 1982, 2016, Oracle.  All rights reserved.


Connected to:
Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production

SQL> ALTER SESSION SET CONTAINER=CDB$ROOT;

Session altered.

SQL> ALTER SYSTEM SET max_string_size=extended SCOPE=SPFILE;

System altered.

SQL> shutdown immediate;
Database closed.
Database dismounted.
ORACLE instance shut down.
SQL> startup upgrade;
ORACLE instance started.

Total System Global Area 1610612736 bytes
Fixed Size		    8793304 bytes
Variable Size		  520094504 bytes
Database Buffers	 1073741824 bytes
Redo Buffers		    7983104 bytes
Database mounted.
Database opened.
SQL> ALTER PLUGGABLE DATABASE ALL OPEN UPGRADE;
EXIT;
Pluggable database altered.

SQL>
Disconnected from Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production
[oracle@285659826a0c ~]$ cd $ORACLE_HOME/rdbms/admin
[oracle@285659826a0c admin]$ mkdir /tmp/log_utl32k
<l/bin/perl $ORACLE_HOME/rdbms/admin/catcon.pl -u SYS -d $ORACLE_HOME/rdbms/admin -l '/tmp/log_utl32k' -b log_utl32k utl32k.sql
catcon: ALL catcon-related output will be written to [/tmp/log_utl32k/log_utl32k_catcon_2370.lst]
catcon: See [/tmp/log_utl32k/log_utl32k*.log] files for output generated by scripts
catcon: See [/tmp/log_utl32k/log_utl32k_*.lst] files for spool files, if any
Enter Password:
catcon.pl: completed successfully
[oracle@285659826a0c admin]$ sqlplus / as sysdba

SQL*Plus: Release 12.2.0.1.0 Production on Tue May 30 19:46:48 2017

Copyright (c) 1982, 2016, Oracle.  All rights reserved.


Connected to:
Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production

SQL> shutdown immediate;
Database closed.
Database dismounted.
ORACLE instance shut down.
SQL> startup;
ORACLE instance started.

Total System Global Area 1610612736 bytes
Fixed Size		    8793304 bytes
Variable Size		  520094504 bytes
Database Buffers	 1073741824 bytes
Redo Buffers		    7983104 bytes
Database mounted.
Database opened.
SQL> ALTER PLUGGABLE DATABASE ALL OPEN READ WRITE;

Pluggable database altered.

SQL> exit;
Disconnected from Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production
[oracle@285659826a0c admin]$ cd $ORACLE_HOME/rdbms/admin
[oracle@285659826a0c admin]$ mkdir /tmp/utlrp
<l/bin/perl $ORACLE_HOME/rdbms/admin/catcon.pl -u SYS -d $ORACLE_HOME/rdbms/admin -l '/tmp/utlrp' -b utlrp utlrp.sql
catcon: ALL catcon-related output will be written to [/tmp/utlrp/utlrp_catcon_2802.lst]
catcon: See [/tmp/utlrp/utlrp*.log] files for output generated by scripts
catcon: See [/tmp/utlrp/utlrp_*.lst] files for spool files, if any
Enter Password:
catcon.pl: completed successfully
[oracle@285659826a0c admin]$ sqlplus / as sysdba

SQL*Plus: Release 12.2.0.1.0 Production on Tue May 30 19:49:28 2017

Copyright (c) 1982, 2016, Oracle.  All rights reserved.


Connected to:
Oracle Database 12c Enterprise Edition Release 12.2.0.1.0 - 64bit Production

SQL> create table test (name varchar2(32676));

Table created.

SQL> alter session set container=ORCLPDB1;

Session altered.

SQL> create table test_pdb (name varchar2(32676));

Table created.

SQL>
```

**Comment by CoreyKaylor** [comment]
Followed your steps I'm pretty sure verbatim and still getting the same errors.

```
[oracle@d731ba648676 bin]$ sqlplus / as sysdba

SQL*Plus: Release 12.1.0.2.0 Production on Wed May 31 01:26:51 2017

Copyright (c) 1982, 2014, Oracle.  All rights reserved.


Connected to:
Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
With the Partitioning, Oracle Label Security, OLAP, Advanced Analytics
and Real Application Testing options

SQL> alter session set container=CDB$ROOT;

Session altered.

SQL> alter system set max_string_size=EXTENDED SCOPE=SPFILE;

System altered.

SQL> shutdown immediate;
Database closed.
Database dismounted.
ORA-00600: internal error code, arguments: [krsh_fsga_sgaq.ds_not_found], [], [], [], [], [], [], [], [], [], [], []
SQL> startup upgrade;
ORACLE instance started.

Total System Global Area 3019898880 bytes
Fixed Size		    2928968 bytes
Variable Size		  738201272 bytes
Database Buffers	 2264924160 bytes
Redo Buffers		   13844480 bytes
Database mounted.
Database opened.
SQL> alter pluggable database all open upgrade;

Pluggable database altered.

SQL> exit;
Disconnected from Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
With the Partitioning, OLAP, Advanced Analytics and Real Application Testing options
[oracle@d731ba648676 bin]$ cd $ORACLE_HOME/rdbms/admin
[oracle@d731ba648676 admin]$ mkdir /tmp/log_utl32k
[oracle@d731ba648676 rdbms]$ cd $ORACLE_HOME
[oracle@d731ba648676 dbhome_1]$ cd rdbms/admin
[oracle@d731ba648676 admin]$ $ORACLE_HOME/perl/bin/perl ./catcon.pl -u SYS -d ./ -l '/tmp/log_utl32k' -b log_utl32k utl32k.sql
catcon: ALL catcon-related output will be written to /tmp/log_utl32k/log_utl32k_catcon_3492.lst
catcon: See /tmp/log_utl32k/log_utl32k*.log files for output generated by scripts
catcon: See /tmp/log_utl32k/log_utl32k_*.lst files for spool files, if any
Enter Password:
catcon.pl: completed successfully
[oracle@d731ba648676 admin]$ cd ..
[oracle@d731ba648676 rdbms]$ cd ..
[oracle@d731ba648676 dbhome_1]$ cd bin
[oracle@d731ba648676 bin]$ sqlplus / as sysdba

SQL*Plus: Release 12.1.0.2.0 Production on Wed May 31 01:37:02 2017

Copyright (c) 1982, 2014, Oracle.  All rights reserved.


Connected to:
Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
With the Partitioning, OLAP, Advanced Analytics and Real Application Testing options

SQL> shutdown immediate;
Database closed.
Database dismounted.
ORACLE instance shut down.
SQL> startup;
ORACLE instance started.

Total System Global Area 3019898880 bytes
Fixed Size		    2928968 bytes
Variable Size		  738201272 bytes
Database Buffers	 2264924160 bytes
Redo Buffers		   13844480 bytes
Database mounted.
Database opened.
SQL> alter pluggable database all open read write;

Pluggable database altered.

SQL> exit;
Disconnected from Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
With the Partitioning, OLAP, Advanced Analytics and Real Application Testing options
[oracle@d731ba648676 bin]$ cd ..
[oracle@d731ba648676 dbhome_1]$ cd rdbms/admin/
[oracle@d731ba648676 admin]$ mkdir /tmp/utlrp
[oracle@d731ba648676 admin]$ $ORACLE_HOME/perl/bin/perl ./catcon.pl -u SYS -d $ORACLE_HOME/rdbms/admin -l '/tmp/utlrp' -b utlrp utlrp.sql
catcon: ALL catcon-related output will be written to /tmp/utlrp/utlrp_catcon_3915.lst
catcon: See /tmp/utlrp/utlrp*.log files for output generated by scripts
catcon: See /tmp/utlrp/utlrp_*.lst files for spool files, if any
Enter Password:
catcon.pl: completed successfully
[oracle@d731ba648676 admin]$ sqlplus / as sysdba

SQL*Plus: Release 12.1.0.2.0 Production on Wed May 31 01:42:23 2017

Copyright (c) 1982, 2014, Oracle.  All rights reserved.


Connected to:
Oracle Database 12c Enterprise Edition Release 12.1.0.2.0 - 64bit Production
With the Partitioning, Oracle Label Security, OLAP, Advanced Analytics
and Real Application Testing options

SQL> create table test (name varchar2(32676));
create table test (name varchar2(32676))
                                 *
ERROR at line 1:
ORA-00910: specified length too long for its datatype

SQL>
```

Container was freshly created with the following.

```
docker run -d --env-file ~/Projects/oracle.env -p 1521:1521 -p 5500:5500 -it --name oracle12c-hm --shm-size="4g" store/oracle/database-enterprise:12.1.0.2
```

With the following env file. Everything else commented out using the defaults.

```
DB_BUNDLE=extreme
```

Docker version 17.03.1-ce, build c6d412e
